src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        LocalDate fromDate = LocalDate.parse(dateFrom,DATE_FORMAT);
+        LocalDate toDate = LocalDate.parse(dateTo, DATE_FORMAT);
+        if (data.length > 0) {
+            String reportPeriod = "Report for period " + dateFrom + " - " + dateTo;
+            result.insert(0, reportPeriod);
+            for (String name : names) {
+                StringBuilder nameResult = new StringBuilder();
```

Comment:
dont create extra StringBuilder, write straight to the existing StringBuilder instead
```suggestion
```